### PR TITLE
Rework entity properties

### PR DESCRIPTION
**Pull request by edk0**
Replace the current system (which calculates on attribute access) with one that pre-calculates attributes. This makes attribute access faster, and with any luck decreases the overall number of times we need to
recalculate values.

This is not quite ready to be pulled—I'm still looking at ways of avoiding the code generation—but I think it's ready to be considered.

### CODE REVIEW

File: fireplace/aura.py [hunk 1]
```diff
@@ -7,10 +7,21 @@ def __init__(self, source, entity):
 		self.source = source
 		self.entity = entity
 		self.tags = CardManager(self)
+		self.recalc = set()
 
 	def __repr__(self):
 		return "<AuraBuff %r -> %r>" % (self.source, self.entity)
 
+	def __setattr__(self, k, v):
+		from .entity import CalculatedProperty
```
Comment (edk0) [inline fireplace/aura.py]: I'm not happy about this

**Comment by djkaty** [comment]
This cannot be ported to Cython.

Is there any reason why these buffs can't be re-calculated on demand without using compilation and dynamic attributes?

File: fireplace/aura.py (hunk 1 above)
Comment (jleclanche) [inline fireplace/aura.py]: So move the code to `property.py` :)

**Comment by jleclanche** [comment]
@edk0 ping - I'd really like to land the version we discussed. When will you have time to work on it?

**Comment by shinoi2** [comment]
Too old and has conflict, close it.

**Pull request closed without merging**